File: challenges/3-references/3-template.java
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.program.model.symbol.*;
import ghidra.program.model.address.*;

public class GetRefs extends GhidraScript {

    @Override
    protected void run() throws Exception {
        String functionName = "flag";
        Address entryPoint;
        Reference[] refs;

        FunctionManager fm = getCurrentProgram().getFunctionManager();
        Function[] functions = fm.getFunctions(true);

        for (Function function : functions) {
            if function.getName().equals(functionName) {
                /*
                entryPoint = ????; // get the entry point of the function 

                refs = ?????; // get the references to entryPoint
                */
                break;
            }
        }
        for (Reference reference : refs) {
            println("Reference to address: 0x" + reference.getFromAddress().getOffset());
        }
    }
}

File: challenges/3-references/3-template.py
import ghidra.app.script.GhidraScript

functionName = "flag" 

fm = currentProgram.getFunctionManager() 
functions = fm.getFunctions(True) 

for func in functions:
    if func.getName() == functionName:
        
        """
        entryPoint = ???? # get the entry point of the function 

        refs = ????? # get the references to entryPoint
        """

        break

for ref in refs:
    print("Reference to address offset: 0x" + str(ref.getFromAddress()))
